pkg/admission/pod_eviction.go

```diff
@@ -0,0 +1,464 @@
+// Provenance-includes-location: https://github.com/kubernetes/kubernetes
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Kubernetes Authors.
+
+package admission
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/spanlogger"
+	v1 "k8s.io/api/admission/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/grafana/rollout-operator/pkg/zpdb"
+)
+
+const (
+	PodEvictionWebhookPath = "/admission/pod-eviction"
+
+	eviction = "eviction"
+
+	// labels we use in logging
+	logMsg   = "msg"
+	logAllow = "pod eviction allowed"
+	logDeny  = "pod eviction denied"
+
+	errUnableToFindPodByName      = "unable to find pod by name"
+	errPodNotReady                = "pod is not ready"
+	errNoPdbForPod                = "no zone pod disruption budgets found for pod"
+	errMaxUnavailableIs0          = "max unavailable = 0"
+	errUnableToDetermineStsStatus = "unable to determine pod status in related StatefulSet"
```

Comment:
Most of these seem to only be used in one place - I wouldn't bother creating constants in that case.